Assets/_Challenges/Scripts/AI Behaviours/Movement/NavMesh/MoveTo.cs

```diff
@@ -0,0 +1,43 @@
+using UnityEngine;
```

Comment:
I'm not sure we want to tie this directly to a navmesh at least for the base behaviour. I think we should create an interface for movement controls and have the agent provide its movement controller. The default for the controller can be navmesh. The abstraction allows us to use a* pro or any other means to actually move the agent.

Reply:
Agreed, that's why this is in a `NavMesh` subfolder. But is that a blocker on this PR? This feels more like a refactor if/when someone needs it. Right now my use case requires a NavMesh and thus I'm implementing that. I propose we have a scratch your own itch policy.

That said, if you feel this would be an overcomplicated refactoring then can you expand a little on what you feel the design should be?

Reply:
Just saw your final note on this to discuss...

Reply:
For future optimization, you may want to have the sampleposition and setdestination methods only be called if targetposition has changed. especially if tick will be called every frame.

Reply:
I agree with @Gildar76 on this. The navmesh sample kills FPS. @Gildar76 in case you missed it, this is not done every frame, but still I like your idea to not poll if the target has not moved.

Reply:
I also agree. This requires a little refactoring as the position variable will vary from agent to agent and thus we can't keep it in the ScriptableObject. We could write it to the Chalkboard but that is quite expensive. Probably the best way to go would be to create a per agent instance of the SO during the Initialize method, then make modifications here. 

The downside I see of this is that it would be in-memory only and thus would be problematic in save games. This can be addressed fairly easily by writing the SO instances to saved game data, but since these values are derived from the environment it might never matter... something to bear in mind.

Anyway, I think this is important enough to do before merge so I will add it shortly.

Reply:
Done the caching of the target location in order to avoid unnecessary NavMesh operations.